### ISSUE TITLE
[BUG] Unable to create vm network, keeps in Not ready state with error cluster network "" not exists

### ISSUE THREAD

**Issue by TachunLin** [harvester/harvester#3579]
**Describe the bug**

When create a vlan network on the vm network page. 
The VM network keep in `Not Ready` state with error message 
```
Cluster Network "" does not exist
```

![image](https://user-images.githubusercontent.com/29251855/222386821-13d81e56-0ba5-4c2c-90d1-c5678b13754e.png)

**To Reproduce**
1. Create a cluster network `cn`
2. Create a network config `nc` and set uplink interface to `ens1f1` 
3. Create a vlan network with id `1216` and dhcp route 
4. Check the vlan network state

**Expected behavior**

Can create vlan network correctly on the vm network page 

**Support bundle**

[supportbundle_dde454bb-56e4-4b47-9d48-d6f9db912b8a_2023-03-02T09-30-26Z.zip](https://github.com/harvester/harvester/files/10869760/supportbundle_dde454bb-56e4-4b47-9d48-d6f9db912b8a_2023-03-02T09-30-26Z.zip)

**Environment**
 - Harvester ISO version: master-616d751f-head (03/01)
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): 4 nodes Harvester on bare machine

**Comment by yaocw2020** [comment]
Reason analysis: 
When creating a new VLAN-type VM network, the harvester-network-controller-manager will update the net-attach-def to add labels to indicate the cluster network, VLAN ID and network type. However, we could find the following error message. 
```
I0302 16:31:32.527641       1 controller.go:103] nad configuration default/issue3579 has been changed: {"cniVersion":"0.3.1","name":"issue3579","type":"bridge","bridge":"demo-br","promiscMode":true,"vlan":1,"ipam":{}}
time="2023-03-02T16:31:32Z" level=error msg="error syncing 'default/issue3579': handler harvester-network-manager-nad-controller: ensure labels of nad default/issue3579 failed, error: Internal error occurred: failed calling webhook \"validator.harvester-system.harvester-network-webhook\": failed to call webhook: Post \"https://harvester-network-webhook.harvester-system.svc:443/v1/webhook/validation?timeout=10s\": EOF, handler harvester-network-manager-nad-controller: Internal error occurred: failed calling webhook \"validator.harvester-system.harvester-network-webhook\": failed to call webhook: Post \"https://harvester-network-webhook.harvester-system.svc:443/v1/webhook/validation?timeout=10s\": EOF, requeuing"
``` 
It indicates that API server failed to call the validating webhook. The EOF error tells us that the webhook panics. Through checking the changed codes, it's obvious that we ignore handling the case where nodeFilter is nil.

**Comment by TachunLin** [comment]
Verified **fixed** on `master-2f37e2be-head`(03/07). Close this issue.

### Result
We can create vlan network in DHCP and Manual mode to route correctly on the VM Networks page
![image](https://user-images.githubusercontent.com/29251855/223316304-aa4725c5-debb-4931-96c5-4d6eac293b5a.png)

### Test Information
* Test Environment: 4 nodes Harvester bare machines
* Harvester version: `master-2f37e2be-head`(03/07)

### Verify Steps

1. Create a cluster network cn
2. Create a network config nc and set uplink interface to ens1f1
3. Create a vlan network with id 1216 and dhcp route
4. Check the vlan network 1216 state
5. Create a vlan network with id 1215 and set manual route CIDR and gateway
6. Check the vlan network 1215 state
7. Delete vlan 1216 and recreate again
